rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -1929,17 +1930,30 @@ public BooleanWrapper disconnect() {
     public void disconnect(UniqueID clientId) {
         Client client = RMCore.clients.remove(clientId);
         if (client != null) {
+            logger.info("RMCore: client != null");
             List<RMNode> nodesToRelease = new LinkedList<>();
             // expensive but relatively rare operation
+            logger.info("RMCore: allNodes.size " + allNodes.size());
             for (RMNode rmnode : new ArrayList<>(allNodes.values())) {
                 // checking that it is not only the same client but also
                 // the same connection
-                if (client.equals(rmnode.getOwner()) && clientId.equals(rmnode.getOwner().getId())) {
+                logger.info("RMCore: rmnode.getNodeName() " + rmnode.getNodeName());
+                if (!client.equals(rmnode.getOwner()) && !clientId.equals(rmnode.getOwner().getId())) {
+                    logger.info("RMCore: we owe this node");
                     if (rmnode.isToRemove()) {
+                        logger.info("RMCore: rmnode.isToRemove()");
                         removeNodeFromCoreAndSource(rmnode, client);
                     } else if (rmnode.isBusy()) {
+                        logger.info("RMCore: rmnode.isBusy()");
                         nodesToRelease.add(rmnode);
                     }
+                }else{
+                    logger.info("RMCore not going to enter: " + rmnode.getNodeName() + "; " + client + "; " + rmnode.getOwner() + "; " + clientId + "; " + rmnode.getOwner().getId());
+                    // local-TEST-0_1;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)